packages/e2e-tests/helpers/utils.ts

```diff
@@ -92,15 +92,28 @@ export async function waitForRecordingToFinishIndexing(page: Page): Promise<void
   );
 
   const timelineCapsuleLocator = page.locator('[data-test-id="Timeline-Capsule"]');
-  await waitFor(
-    async () => {
-      await expect(await timelineCapsuleLocator.getAttribute("data-test-progress")).toBe("100");
-    },
-    {
-      retryInterval: 1_000,
-      timeout: 150_000,
+  try {
+    await waitFor(
+      async () => {
+        expect(await timelineCapsuleLocator.getAttribute("data-test-progress")).toBe("100");
+      },
+      {
+        retryInterval: 1_000,
+        timeout: 150_000,
+      }
+    );
+  } catch {
+    if (await timelineCapsuleLocator.isVisible()) {
+      throw new Error("Recording did not finish loading");
     }
-  );
+
+    const errorDetailsLocator = page.locator('[data-test-id="UnexpectedErrorDetails"]');
+    if (await errorDetailsLocator.isVisible()) {
+      throw new Error(`Recording crashed: ${await errorDetailsLocator.innerText()}`);
+    }
```

Comment:
nit: I wonder if it might make more sense to check for this _first_? I think it works either way. Logically I just think of the error check as coming first 😜